src/redemptions/redemption.service.ts

```diff
@@ -246,6 +255,34 @@ export class RedemptionService {
     return { eligible: true, reason: '' };
   }
 
+  currentDate = (): Date => {
+    return new Date();
+  };
+
+  userDeadline = async (userId: number): Promise<Date> => {
+    const userPoints = await this.userPointsService.findOrThrow(userId);
+    const eligiblePools = [];
+    const pool1 = AIRDROP_CONFIG.data.find((c) => c.name === 'pool_one');
+    if (pool1 && userPoints.pool1_points !== 0) {
+      eligiblePools.push(pool1);
+    }
+    const pool2 = AIRDROP_CONFIG.data.find((c) => c.name === 'pool_two');
+    if (pool2 && userPoints.pool2_points !== 0) {
+      eligiblePools.push(pool2);
+    }
+    const pool3 = AIRDROP_CONFIG.data.find((c) => c.name === 'pool_three');
+    if (pool3 && userPoints.pool3_points !== 0) {
+      eligiblePools.push(pool3);
+    }
+    const pool4 = AIRDROP_CONFIG.data.find((c) => c.name === 'pool_four');
+    if (pool4 && userPoints.pool4_points !== 0) {
+      eligiblePools.push(pool4);
+    }
+    return new Date(
```

Comment:
Note this will return `-Inf` if no eligible pools are present for a `userId`

Reply:
We can definitely handle this explicitly, but user can't  create redemption if they don't have points in one of phases.

Reply:
Yeah, I think it's fine as is. Just was a note while reading this